src/NTDll/NTDll.cs

```diff
@@ -70,6 +70,34 @@ public static partial class NTDll
             Kernel32.VER_MASK TypeMask,
             long ConditionMask);
 
+        /// <summary>
+        /// Retrieves information about the specified process.
+        /// </summary>
+        /// <param name="ProcessHandle">
+        /// A handle to the process for which information is to be retrieved.
+        /// </param>
+        /// <param name="ProcessInformationClass">
+        /// The type of process information to be retrieved.
+        /// </param>
+        /// <param name="ProcessInformation">
+        /// A pointer to a buffer supplied by the calling application into which the function writes the requested information.
+        /// </param>
+        /// <param name="ProcessInformationLength">
+        /// The size of the buffer pointed to by the <paramref name="ProcessInformation"/> parameter, in bytes.
+        /// </param>
+        /// <param name="ReturnLength">
+        /// A pointer to a variable in which the function returns the size of the requested information. If the function
+        /// was successful, this is the size of the information written to the buffer pointed to by the
+        /// ProcessInformation parameter, but if the buffer was too small, this is the minimum size of buffer needed
+        /// to receive the information successfully.
+        /// </param>
+        /// <returns>
+        /// The function returns an NTSTATUS success or error code.
+        /// </returns>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/desktop/ms684280(v=vs.85).aspx"/>
+        [DllImport(nameof(NTDll))]
+        public static extern NTSTATUS NtQueryInformationProcess(Kernel32.SafeObjectHandle ProcessHandle, PROCESSINFOCLASS ProcessInformationClass, ref PROCESS_BASIC_INFORMATION ProcessInformation, int ProcessInformationLength, IntPtr ReturnLength);
```

Comment:
* `ReturnLength` should be `out int` instead of `IntPtr`. I use `int` instead of `uint` here because many .NET APIs use `int` instead, and it's inconceivable that Windows would want to write a value here in excess of `int.MaxValue`.
* `ProcessInformation` as you have it here is typed for a specific struct which only sometimes applies based on `ProcessInformationClass`. We should use `void*` here instead, I think. You can add a `[Friendly(FriendlyFlags.Out)]` parameter to it perhaps, although I'm not sure it will work for `void*`. Ultimately the caller will have to cast that `void*` to `PROCESS_BASIC_INFORMATION*` if they expect that to be the relevant struct (or use a `Marshal` method to copy the data into a struct).

Reply:
You're correct that the type of the result stored in `ProcessInformation` will vary based on the the value of `ProcessInformationClass`. Would be an option to have multiple overloads of NtQueryInformationProcess for the different output types?

Reply:
We prefer to have just one method defined that matches the signature found in the header file.

Reply:
OK, I updated the code accordingly. Thanks!